packages/next/src/lib/metadata/default-metadata.tsx

```diff
@@ -1,12 +1,17 @@
 import React from 'react'
 import type { ResolvedMetadata } from './types/metadata-interface'
 
-export const createDefaultMetadata = (): ResolvedMetadata => {
+export function createDefaultMetadata(): ResolvedMetadata {
+  const defaultMetadataBase =
+    process.env.NODE_ENV === 'production' && process.env.VERCEL_URL
+      ? new URL(`https://${process.env.VERCEL_URL}`)
+      : null
```

Comment:
Maybe we should fallback to a different env var like `NEXT_METADATA_URL`?

Reply:
For vercel deployment case we want it just works, but by default for `next export` or self hosting you need to set it explicitly in the code. We consume env here just because vercel provides this env